app/models/template.rb

```diff
@@ -88,9 +88,11 @@ def import_without_save(text, options = {})
     self.template = text
     @importing_metadata = self.class.parse_metadata(text)
     Foreman::Logging.logger('app').debug "setting attributes for #{name} with id: #{id || 'N/A'}"
-    self.snippet = !!@importing_metadata[:snippet]
+    self.snippet = !!@importing_metadata["snippet"]
     self.default = options[:default] unless options[:default].nil?
-    self.description = @importing_metadata[:description]
+    self.description = @importing_metadata["description"]
+    template_kind = TemplateKind.find_by_name(@importing_metadata["kind"])
```

Comment:
Find by name should not be used if not necessary :) Also if the kind is not set or is a snippet, I'd prefer if we do not query the database - it is bit more efficient.
```suggestion
    template_kind = TemplateKind.find_by(name: @importing_metadata["kind"]) if @importing_metadata["kind"] && @importing_metadata["kind"] != 'snippet'
```
might look better not to inline the `if`, but up to you :)